nautobot/extras/forms/forms.py

```diff
@@ -1013,6 +1013,10 @@ class JobHookForm(BootstrapMixin, forms.ModelForm):
     content_types = MultipleContentTypeField(
         queryset=ChangeLoggedModelsQuery().as_queryset(), required=True, label="Content Type(s)"
     )
+    job = DynamicModelChoiceField(
+        queryset=Job.objects.filter(is_job_hook_receiver=True),
+        query_params={"is_job_hook_receiver": True},
+    )
```

Comment:
We were missing this (and the similar addition below) which resulted in the form using the default (server-side-rendered) form field instead.